### DISCUSSION TITLE
Add check for correct case of username

### DISCUSSION THREAD

**Discussion by SecondThundeR** [archlinux/archinstall#168] [Ideas]
Hi! I'm very glad that Arch is finally trying to make a user-friendly installer that it is simple and powerful. But there are small flaws.

When I tried to set up the first user, I entered my username in a non-lower case and completely forgot that this isn't allowed in Linux. However, the script didn't gave me any warnings and continued its work further.
At the end of the installation, after `Enabling service gdm`, I received an SysCallError exception with message `useradd: invalid user name` and the installation was not completed. Here are an example of issue:

![image](https://user-images.githubusercontent.com/36604233/113517247-673b5f80-9587-11eb-8181-5b17720cc11c.png)

I think it would be great to add a little case check in functions like this: [ask_for_superuser_account](https://github.com/archlinux/archinstall/blob/master/archinstall/lib/user_interaction.py#L50) and [ask_for_additional_users](https://github.com/archlinux/archinstall/blob/master/archinstall/lib/user_interaction.py#L65).
For example, we can make our loop endless until the user enters the username correctly:
```python
while 1:
  new_user = input(prompt).strip(' ')
  if not new_user:
    break
  if not new_user.islower():
    prompt = "Invalid username, please enter it in lowercase:"
  else:
    password = get_password(prompt=f'Password for user {new_user}: ')
    # and so on...
```
but this solution adds unnecessary nested if..else branch, so we can raise an exception:
```python
while 1:
  new_user = input(prompt).strip(' ')
  if not new_user:
    break
  if not new_user.islower():
    raise Exception(...)
```

I can't say for sure that these solutions are great for performance or code quality, but I think that this check is needed in the script so that any users who wanted to install Arch in this way wouldn't be confused after a similar error. Hope this helps in some way

**Comment by Torxed** [comment]
Wait, this is going to sound silly but, 20 years of using Linux and I think this is the first time I've ever been pointed out that Linux doesn't support upper-case usernames.. If this is the case (which it most likely is) we should for sure do a check/warning or even default to lower-case the usernames..

I'm transferring this to the issue section and if you wish you're welcome to add a PR and I'll merge it.
I would personally preference just doing `input(prompt).strip(' ').lower()` if that doesn't cause confusion.

Edit: weird, can't transfer this.. I used to be able to transfer discussions to issues oÔ

**Reply by SecondThundeR** [reply]
I'm apologize for writing here, and not in Issue, so I thought it would probably be a silly report for a trivia like this. I will make a PR in a few minutes

**Reply by kpcyrd** [reply]
Also note that linux usernames can't (or at least shouldn't) start with numbers :)

**Comment by Torxed** [comment]
Note to self: This was solved in PR #170. Good job on this one!
